solutions/src/main/java/ru/sberbank/school/task02/ReverseCalculator.java

```diff
@@ -35,63 +38,75 @@
         if (quotes.isEmpty()) {
             throw new FxConversionException("No quotes found");
         }
-        Quote current = null;
-        for (Quote quote: quotes) {
-            BigDecimal price = operation == ClientOperation.BUY ? quote.getBid() : quote.getOffer();
-            BigDecimal reverseAmount = amount.divide(price, 10, RoundingMode.HALF_UP);
-            current = calculator.checkQuote(reverseAmount, quote, current);
-        }
-        if (current == null) {
+
+        List<Quote> filteredQuotes = filterQuotes(amount, quotes, operation);
+
+        if (filteredQuotes.isEmpty()) {
             return Optional.empty();
         }
-        BigDecimal priceResult = operation == ClientOperation.BUY ? current.getBid() : current.getOffer();
-        BigDecimal revertResult = BigDecimal.ONE.divide(priceResult, 10, RoundingMode.HALF_UP);
-        return Optional.ofNullable(revertResult);
+        BigDecimal priceResult;
+        if (filteredQuotes.size() == 1) {
+            Quote current = filteredQuotes.get(0);
+            priceResult = operation == ClientOperation.BUY ? current.getBid() : current.getOffer();
+         } else {
```

Comment:
&#39;if rcurly&#39; has incorrect indentation level 9, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a873371774>*